File: leetcode/python/2028.find-missing-observations.py
class Solution:
    def missingRolls(self, rolls: List[int], mean: int, n: int) -> List[int]:
        rem = (len(rolls) + n) * mean - sum(rolls)
        if rem > n * 6 or rem < n:
            return []
        ans = [rem//n] * n
        rem = rem % n
        for i in range(rem):
            ans[i] += 1
        return ans

File: leetcode/cpp/2028.find-missing-observations.cpp
class Solution {
public:
    vector<int> missingRolls(vector<int>& rolls, int mean, int n) {
        int rem = (n + rolls.size()) * mean;
        for (const int roll: rolls) {
            rem -= roll;
        }
        if (rem > n * 6 || rem < n) {
            return {};
        }
        vector<int> ans(n, rem/n);
        rem = rem % n;
        for (int i = 0; i < rem; i++) {
            ans[i]++;
        }
        return ans;
    }
};

